components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts

```diff
@@ -1,139 +1,80 @@
-      this.projectsUpdated.subscribe(() => {
-        this.updateLabel();
-      });
-    }
-  }
-
-  ngOnChanges(changes: SimpleChanges): void {
-    this.updateLabel();
-    // only update the projects list on initialization/the first change
-    if (changes.projects && changes.projects.firstChange) {
-      this.filteredProjects = this.projectsArray;
-    }
-  }
+  // Emits checked set of ids upon completion.
+  @Output() onDropdownClosing = new EventEmitter<string[]>();
 
-  get projectsArray(): ProjectChecked[] {
-    return ChefSorters.naturalSort(Object.values(this.projects), 'name');
-  }
+  // Label to use when none are selected
+  public noneSelectedLabel = ProjectConstants.UNASSIGNED_PROJECT_ID;
 
-  toggleDropdown(event: MouseEvent): void {
-    event.stopPropagation();
-    if (this.disabled) {
-      return;
-    }
-    if (!this.active) {
-      this.filterValue = '';
-      this.filteredProjects = this.projectsArray;
-    }
+  // plural display name of resource
+  public objectNounPlural = 'projects';
 
-    this.active = !this.active;
-  }
+  public projects: ResourceChecked[];
 
-  projectChecked(checked: boolean, project: ProjectChecked): void {
-    project.checked = checked;
-    this.updateLabel();
-    this.onProjectChecked.emit(project);
-  }
+  private isDestroyed = new Subject<boolean>();
 
-  closeDropdown(): void {
-    if (this.active) {
-      this.active = false;
-    }
-  }
+  constructor(private store: Store<NgrxStateAtom>) { }
 
-  handleFilterKeyUp(): void {
-    this.filteredProjects = this.filterProjects(this.filterValue);
+  ngOnInit(): void {
+    this.store.select(assignableProjects)
+      .pipe(takeUntil(this.isDestroyed))
+      .subscribe((assignable: ProjectsFilterOption[]) => {
+        this.projects =
+          assignable.map(p => {
+            return <ResourceChecked>{
+              id: p.value,
+              name: p.label,
+              type: p.type,
+              checked: this.checkedProjectIDs
+                && this.checkedProjectIDs.includes(p.value)
+            };
+          });
+      });
   }
 
-  filterProjects(value: string): ProjectChecked[]  {
-    return this.projectsArray.filter(project =>
-      project.id.toLowerCase().indexOf(value.toLowerCase()) > -1
-    );
+  ngOnDestroy() {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
   }
 
-  moveFocus(event: KeyboardEvent): void {
-    event.preventDefault();
-    let nextElement: HTMLElement;
-
-    const targetElement = <Element>event.target;
-    if (event.key === 'ArrowUp') {
-      nextElement = <HTMLElement>targetElement.previousElementSibling;
-    } else if (event.key === 'ArrowDown') {
-      nextElement = <HTMLElement>targetElement.nextElementSibling;
-    }
-
-    if (nextElement == null) {
-      return;
-    } else {
-      nextElement.focus();
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.checkedProjectIDs && changes.checkedProjectIDs.currentValue && this.projects) {
```

Comment:
There are two sets of inputs to be concerned with: the set of all existing projects and the set of projects previously associated with the token. The block above generates the projects list from the ngrx state representation of allowable projects. Here we factor in the set of projects the token had previously selected (if any). Think of `onChanges` like an Observable for `@Input`s to the component.